File: adventofcode/2021/day2/day2.cpp
#include "iostream"
#include "fstream"
#include "sstream"

using namespace std;

int main(){
    

    string dir;
    int horiz = 0;
    int depth = 0;
    int aim = 0;
    while(cin>>dir){
        int x;
        cin>>x;
        if(dir == "forward"){
            horiz += x;
            depth += aim * x;
        }else if(dir == "down"){
            aim += x;
        } else if (dir == "up") {
            aim -= x;
        }
    }   
    
    cout << depth * horiz << endl;


}

File: 2021/day2/day2.py


from os import read


def answer():
    f = open("input.txt", "r")
    rawline = f.readline()
    print(rawline)
    horiz = 0
    depth = 0
    aim = 0
    while(rawline != ""):
        line = rawline.split()
        print(line)
        direction = line[0]
        if(direction == "forward"):
            horiz += int(line[1])
            depth += aim * int(line[1])
        elif(direction == "down"):
            aim += int(line[1])
        elif(direction == "up"):
            aim -= int(line[1])
        rawline = f.readline()

    print(horiz, depth)
    print(horiz * depth)
        

        



if __name__ == "__main__":
    print(answer())
